quantum/rgblight.c

```diff
@@ -137,33 +135,9 @@ void setrgb(uint8_t r, uint8_t g, uint8_t b, LED_TYPE *led1) {
 }
 
 void rgblight_check_config(void) {
-  /* Add some out of bound checks for RGB light config */
-
-  if (rgblight_config.mode < RGBLIGHT_MODE_STATIC_LIGHT) {
```

Comment:
Why doesn't this function check the mode value any more? Surely you need to cap against the maximum permissible mode?

Reply:
I was wondering that myself.  However, the answer is actually simple, it's calling this when the values change: 
https://github.com/qmk/qmk_firmware/blob/c98247e3dd2958bd2d8969dc75170e7e2757b895/lib/lib8tion/math8.h#L23-L28

So, if that is the case, this actually handles that, so a manual check isn't needed, in theory.

Reply:
Hmmm, sure for when values are being incrementally adjusted (BTW, why do apparently unused `rgblight_increase` and `rgblight_decrease` even exist when there are the `rgblight_step*` functions?). But what about if the mode value stored in `rgblight_config` as read out of eeprom or potentially as transferred between halves of a split keyboard is out of range compared to what the firmware thinks the number of modes are? (e.g. the `rgblight_config` struct change here allocates an extra bit to the mode value, so the value read out will be different to what was originally written). 

Also potentially relevant is #5336

Reply:
That was a mistake, mode checks should remain as its range is less than uint8_t range. Will fix

Reply:
Fixed, if #5336 lands before this pr, will update this pr at that time.